docs/INSTALL.rst

```diff
@@ -54,3 +54,19 @@ However if you prefer, you may install ``teuthology`` from `PyPI <http://pypi.py
 
 **Note**: The version in PyPI can be (*far*) behind the development version.
 
+Or from GitHub::
+
+    pip install git+https://github.com/ceph/teuthology#egg=teuthology[orchestra]
+
+where the dependencies for orchestrating are installed. They are used for
+interacting with the services to schedule tests and to report the test results.
+
+
+Update Dependencies
+-------------------
+
+We track the dependencies using ``requirements.txt``. These packages are tested,
+and should work with teuthology. But if you want to bump up the versions of them,
+please use the following command to update this file ::
+
+  pip-compile -qo- | sed '/^-e / d' > requirements.txt
```

Comment:
Instead of this recipe and creating requirements.in, what about providing a wrapper (as supported by `pip-compile`) which also embeds extras?:
```sh
# file: update-requirements.sh, allows for 2 input env vars: $PYTHON_VERSION and $EXTRAS
PYTHON_VERSION=${PYTHON_VERSION:-3}
EXTRAS=${EXTRAS:-coverage,orchestra,test}
$ CUSTOM_COMPILE_COMMAND="PYTHON_VERSION=${PYTHON_VERSION} EXTRAS=${EXTRAS} generate-requirements.sh" python${PYTHON_VERSION} -mpiptools compile -qo- <(echo "-e .[${EXTRAS}]") | sed '/^-e / d' > requirements-py${PYTHON_VERSION}.txt
``` 

It would generate a requirements-py{27,3}.txt like this:
```sh
# This file is autogenerated by pip-compile
# To update, run:
#
#    PYTHON_VERSION=3 EXTRAS=coverage,orchestra,test generate-requirements.sh
#
```

Reply:
you would need a requirements file for every supported python3 version because the output of `pip-compile` might be different for different python3 minor versions...